### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by shm04**
### In this pull request I add the following changes:

- Add the following validations:
   - For the User model:
       - Name must not be blank.
       - PostsCounter must be an integer greater than or equal to zero.
   - For the Post model:
      - Title must not be blank.
      - Title must not exceed 250 characters.
      - CommentsCounter must be an integer greater than or equal to zero.
       - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (cynthiainga) [inline spec/models/user_spec.rb:1]: - [ ] Good job so far but as mentioned in the project requirement all of your models' methods should be tested, kindly add the test for `recent_posts` method 🙃

File: spec/models/post_spec.rb
Comment (cynthiainga) [inline spec/models/post_spec.rb:1]: - [ ] Good job so far but as mentioned in the project requirement all of your models' methods should be tested, kindly add the test for `recent_comments` and `update_user_posts_counter` methods 🙃 🙃

File: spec/models/like_spec.rb
Comment (cynthiainga) [inline spec/models/like_spec.rb:1]: - [ ] Good job so far but as mentioned in the project requirement all of your models' methods should be tested, kindly add the test for the `update_post_likes_counter` method 😁

File: spec/models/comment_spec.rb
Comment (cynthiainga) [inline spec/models/comment_spec.rb:1]: - [ ] Good job so far but as mentioned in the project requirement all of your models' methods should be tested, kindly add the test for the `update_post_comments_counter` method 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @shm04, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] All implemented tests are passing ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @shm04    :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**